src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -146,6 +175,7 @@ export const migrations = {
         throw new Error(`Failure attempting to migrate saved object '${doc.attributes.title}' - ${e}`);
       }
     },
+    '7.0.1': removeDateHistogramTimeZones,
```

Comment:
There should only be one - one of these needs to be removed. It doesn't necessarily have to follow the versioning of Kibana. On startup, it will see if there are any newer versions for the visualization type, we will run those migrations to get up to that version.

Reply:
I think there needs to be 2, because some clients will have run the 7.0.0 migrations already, but some will not have (those running 6.x). And we want to fix both people without forcing the 6.x people to upgrade to 7.x. Does that make sense, @tylersmalley

Reply:
We talked about that offline and it makes sense now to everyone. I updated the code to have a better explanation why we need it twice.